includes/wc-template-functions.php

```diff
@@ -2738,3 +2738,17 @@ function wc_page_noindex() {
 function wc_get_theme_slug_for_templates() {
 	return apply_filters( 'woocommerce_theme_slug_for_templates', get_option( 'template' ) );
 }
+
+/**
+ * Load up unsupported theme enhancements if needed.
+ *
+ * @todo Move this into the main unsupported theme loader.
+ */
+function wc_enhance_unsupported_theme_product_singles( $template ) {
```

Comment:
This function can probably be a method in the template loader class. It's all related IMO.